### ISSUE TITLE
Be clear in plugin contribution docs about `make test` and `make check-style`

### ISSUE THREAD

**Issue by mickmister** [mattermost/mattermost-developer-documentation#943]
The main checks our CI makes are done using the commands `make test` and `make check-style`. Contributors usually don't know about these commands, though they are explained in the plugin dev docs here https://developers.mattermost.com/integrate/plugins/developer-workflow.

We can surface these "especially helpful for contributions" commands on this page https://developers.mattermost.com/contribute/plugins.

**Comment by haardikdharma10** [comment]
Makes sense. Even I was unaware about them until I skimmed through the Makefile. @mickmister as you mentioned above, this should be under the heading "especially helpful for contributions". So do you want this to be added in https://developers.mattermost.com/contribute/plugins as a separate heading or it should link to https://developers.mattermost.com/integrate/plugins/developer-workflow from the plugins page? 

@cwarnermm taking you in loop here is you have any suggestions 🙂

**Comment by mickmister** [comment]
@haardikdharma10 Maybe the action here is to list `make test` and `make check-style` at the top of the https://developers.mattermost.com/integrate/plugins/developer-workflow page, instead of in the middle of the list. This will emphasize the importance of running CI checks locally.

I propose the first five commands be reordered, resulting in this list:

- make test
- make check-style
- make deploy
- make watch
- make dist
- make enable
- make disable
- make reset
- make attach-headless
- make clean

0/5 linking to the developer workflow page provides the full context and encourages reading those pages if we keep the technical explanation bare in the contribution page. The contribution page https://developers.mattermost.com/contribute/plugins is pretty bare in general though, so it could probably use more content such as this. @justinegeffen Any thoughts on this case?

**Comment by cwarnermm** [comment]
Really liking the suggestions and brainstorming on this PR! I recommend documenting the commands in one location, then linking to them from other areas of the documentation based on the user's workflow. Great suggestions on how to position the content, @mickmister!

**Comment by haardikdharma10** [comment]
Thank you for your thoughts and suggestions @mickmister and @cwarnermm. Really appreciate both of your time on this. I'll go ahead and open up a PR and we can discuss there, if there will be any blockers.
